[WIP] Add a script to create catalog source configmap

nikhil-thomas opened the pull request:
Add script to generate tektoncd-operator manifest

Add a sample tektoncd-operator manifest

nikhil-thomas committed:
Add a script to create catalog source configmap

Add tektoncd.package manifest

Add a script to generate a manifest which defines
 - namespace
 - operatorgroup
 - catalogsource
 - configmap
reference:
https://github.com/openshift-cloud-functions/knative-operators/blob/ec60ae6ebb336bd743f98b1015456ac311bfdf55/etc/scripts/catalog.sh

Add a sample tektoncd-operator manifest

Signed-off-by: Nikhil Thomas <nikhilthomas1@gmail.com>

sthaha commented on scripts/catalog_configmap.sh:
```diff
@@ -0,0 +1,59 @@
+
+cat <<EOF | sed 's/^  *$//'
+---
+apiVersion: v1
```
we may want to add comment that this is autogenerated.

nikhil-thomas replied:
added a comment inside the here document to indicate the resulting catalog manifest is autogenerated

nikhil-thomas committed:
Improve olm catalog createion script

Rename olm catalog createion script (new name: olm_catalog.sh)

Modify olm-catalog to add a comment in generated olm-catalog manifest,
to indicate that the catalog was autogenerated

Signed-off-by: Nikhil Thomas <nikhilthomas1@gmail.com>
